plugins/git/git_commit_behind

```diff
@@ -0,0 +1,260 @@
+
+Link this plugin, as usual.
+For example :
+  ln -s /path/to/git_commit_behind /etc/munin/plugins/git_commit_behind
+
+You also need to setup a cron job to trigger the git fetches.
+
+The plugin can be called with an "update" mode to handle the fetches :
+munin-run git_commit_behind update <maxinterval> <probability>
+It will run the fetches randomly (1 in <probability> chances),
+and ensure that it is run at least every <maxinterval> seconds.
+
+For example, you can use the following cron :
+
+# If the git_commit_behind plugin is enabled, fetch git repositories approx.
+# once an hour (12 invocations an hour, 1 in 12 chance that the update will
+# happen), but ensure that there will never be more than two hours
+# (7200 seconds) interval between updates.
+*/5 * * * * root if [ -x /etc/munin/plugins/git_commit_behind ]; then /usr/sbin/munin-run git_commit_behind update 7200 12 >/dev/null; fi
+
+=head1 CONFIGURATION
+
+Use your "/etc/munin/plugin-conf.d/munin-node" to configure this plugin.
+    [git_commit_behind]
+    user root
+    env.git_path /path/to/git
+
+Then, for each repository you want to check, you need the following
+configuration block under the git_commit_behind section
+    env.repo.[repoCode].path /path/to/local/repo
+    env.repo.[repoCode].name Repo Name
+    env.repo.[repoCode].user user
+    env.repo.[repoCode].warning 10
+    env.repo.[repoCode].critical 100
+
+[repoCode] can only contain letters, numbers and underscores.
+
+path : mandatory, the local path to your git repository
+name : optional (default : [repoCode]), a cleaner name that will be displayed
+user : optional (default : empty), the owner of the repository
+       if set and different from the user running the plugin, the git commands
+       will be executed as this user
+warning : optional (default 10), the warning threshold
+critical : optional (default 100), the critical threshold
+
+For example :
+
+    [git_commit_behind]
+    user root
+
+    env.repo.munin_contrib.path /opt/munin-contrib
+    env.repo.munin_contrib.name Munin Contrib
+
+    env.repo.other_repo.path /path/to/other-repo
+    env.repo.other_repo.name Other Repo
+
+=head1 MAGIC MARKERS
+
+  #%# family=auto
+  #%# capabilities=autoconf
+
+=head1 VERSION
+
+1.0.0
+
+=head1 AUTHOR
+
+Neraud (https://github.com/Neraud)
+
+=head1 LICENSE
+
+GPLv2
+
+=cut"""
+
+
+from __future__ import print_function
+
+import logging
+import os
+from random import randint
+import re
+from subprocess import check_output, call, DEVNULL, CalledProcessError
+import sys
+import time
+
+
+plugin_version = "1.0.0"
+
+debug = int(os.getenv('MUNIN_DEBUG', os.getenv('DEBUG', 0))) > 0
+if debug:
+    logging.basicConfig(level=logging.DEBUG,
+                        format='%(asctime)s %(levelname)-7s %(message)s')
+
+conf = {
+    'git_path':  os.getenv('git_path', '/usr/bin/git'),
+    'state_file': os.getenv('MUNIN_STATEFILE',
+                            '/var/lib/munin-node/plugin-state/nobody/' +
```

Comment:
Please rely on the munin node to supply a `MUNIN_STATEFILE` path - otherwise this feature is simply not supported.